client/utils/handleSingleAndDoubleClick.js

```diff
@@ -0,0 +1,30 @@
+import { debounce } from 'lodash'
+
+// Due to limitations of ES6 default strict mode on all files, we cannot bind the
+// 'this' object from calling code. Hence, passing this directly as 'self' (ugly).
```

Comment:
I ran into some issues trying to use bound parameters to the lodash's debounce function. But I guess I got this working fine now. I'm passing callback functions as parameters to this utility function now. React not handling onClick and onDoubleClick on the same component made me use the lodash's debounce workaround. And we thought we'll put this in utility because EventsList might use this too. If this function was a part of the component itself, it would be much clearer. However, I do understand it is quite confusing. Please let me know what you think with the new changes. Thanks :)